### PR TITLE
Precompiler

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: lib/unifex.ex [hunk 1]
```diff
@@ -0,0 +1,43 @@
+defmodule Unifex do
```
Comment (mickel8) [inline lib/unifex.ex:1]: I would place it under `lib/unifex/precompiler.ex` and name this module as `defmodule Unifex.Precompiler` but up to you

Comment (mat-hek) [reply lib/unifex.ex:1]: That way one would have to put `Unifex.Precompiler` there:
https://github.com/membraneframework/unifex/pull/49/files#diff-d4447cfb5dbff840134840900481468bR32

File: lib/unifex/code_generator.ex [hunk 2]
```diff
@@ -45,9 +49,8 @@ defmodule Unifex.CodeGenerator do
   defp get_generator_module_name(interface) do
     module_name =
       case interface do
-        :nif -> :NIF
-        :cnode -> :CNode
-        other -> other
```
Comment (mickel8) [inline lib/unifex/code_generator.ex:50]: This line was for user defined generators as well as for these specified in `*.spec.exs` file. Removing it we will probably support only `:nif` and `:cnode` generators specified in `bundlex.exs` :roll_eyes:

File: lib/unifex.ex (hunk 1 above)
Comment (mickel8) [reply lib/unifex.ex:1]: Oh, I see :+1:

**Pull request merged**